CONTRIBUTING.rst

```diff
@@ -15,91 +15,126 @@ If you are new to git, you can follow one of these tutorials:
+You are all set!
+Now, he basic WarpX development workflow is:
 
-      git checkout development
-      git pull upstream development
+1. Implement your changes and push them on a new branch ``<branch_name>`` on your fork.
+2. Create a Pull Request from branch ``<branch_name>`` on your fork to branch ``development`` on the main WarpX repo.
 
-Make sure you are on WarpX ``development`` branch with
+Create a branch ``<branch_name>`` (the branch name should reflect the piece of code you want to add, like ``fix-spectral-solver``) with
 
 .. code-block:: sh
 
+   # start from an up-to-date development branch
    git checkout development
+   git pull mainline development
 
-in the WarpX directory.
+   # create a fresh branch
+   git checkout -b <branch_name>
 
-Create a branch ``<branch_name>`` (the branch name should reflect the piece of code you want to add, like ``fix_spectral_solver``) with
+and do the coding you want.
 
-.. code-block:: sh
+It is probably a good time to look at the `AMReX documentation <https://amrex-codes.github.io/amrex/docs_html/>`_ and at the Doxygen reference pages:
 
-   git checkout -b <branch_name>
+* WarpX Doxygen: https://warpx.readthedocs.io/en/latest/_static/doxyhtml
+* AMReX Doxygen: https://amrex-codes.github.io/amrex/doxygen
+* PICSAR Doxygen: (todo)
 
-and do the coding you want.
-It is probably a good time to look at the `AMReX documentation <https://amrex-codes.github.io/amrex/docs_html/>`_ and at the `AMReX Doxygen reference <https://ccse.lbl.gov/pub/AMReX_Docs/index.html>`_.
-Add the files you work on to the git staging area with
+Once you are done developing, add the files you modified to the ``git`` *staging area* with
 
 .. code-block:: sh
 
    git add <file_I_created> <and_file_I_modified>
 
+
+Build your changes
+^^^^^^^^^^^^^^^^^^
+
+If you changed C++ files, then now is a good time to test those changes by compiling WarpX locally.
+Follow the `developer instructions in our manual <https://warpx.readthedocs.io/en/latest/install/cmake.html>`__ to set up a local development environment, then compile and `run <https://warpx.readthedocs.io/en/latest/usage/how_to_run.html>`__ WarpX.
+
+
 Commit & push your changes
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Periodically commit your changes with
 
 .. code-block:: sh
 
-   git commit -m "This is a 50-char description to explain my work"
+   git commit
+
+The commit message (between quotation marks) is super important in order to follow the developments during code-review and identify bugs.
+A typical format is:
+
+.. code-block::
+
+   This is a short, 40-character title
+
+   After a newline, you can write arbitray paragraphs. You
+   usually limit the lines to 70 characters, but if you don't, then
+   nothing bad will happen.
 
-The commit message (between quotation marks) is super important in order to follow the developments and identify bugs.
+   The most important part is really that you find a descriptive title
+   and add an empty newline after it.
 
```

Comment:
Here we could add something like: "If you realize that you made a mistake in your commit message and want to change it, before pushing the commit, you can do this with ``git commit --amend``, which will open the commit message in an editor and let you modify it."

Reply:
I think that's too advanced for here. Multiple commits do not hurt for beginners as we squash them on merge.

History rewriting (amend, rebase) are advanced `git` topics and I don't want people to footgun themselves by starting right away with `push -f` when they are first time using `git`.